bound/kt/src/test/kotlin/web5/sdk/dids/methods/web/DidWebTests.kt

```diff
@@ -1,50 +1,276 @@
+            testSuite.include()
+
+            val controllers = listOf("did:web:controller.example.com")
+
+            val bearerDid = DidWeb.create(
+                "https://example.com",
+                DidWebCreateOptions(controller = controllers)
+            )
+
+            assertEquals(controllers, bearerDid.document.controller)
+        }
     }
 
-    @Test
-    fun `can resolve did web uri`() {
-//        Works if you host a local did web document You must host this json at http://localhost:1234/.well-known/did.json
-//        val didDocumentJson = """
-//            {
-//               "id":"did:web.tbd.website",
-//               "@context":[
-//                  "https://www.w3.org/ns/did/v1"
-//               ],
-//               "verificationMethod":[
-//                  {
-//                     "id":"did:web:www.tbd.website#key-0",
-//                     "type":"JsonWebKey",
-//                     "controller":"did:web:www.tbd.website",
-//                     "publicKeyJwk":{
-//                        "alg":"Ed25519",
-//                        "kty":"OKP",
-//                        "crv":"Ed25519",
-//                        "x":"gNFtgCZhOYv00p48FHQYt4edkoBPOyw0oGAB20LrT0c"
-//                     }
-//                  }
-//               ]
-//            }
-//        """.trimIndent()
-//
-//        val didUri = "did:web:localhost%3A1234"
-//        val resolvedDid = DidWeb.resolve(didUri)
-//
-//        assertEquals(resolvedDid.document!!.id, "did:web:www.tbd.website")
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class Resolve {
+        private val testSuite = UnitTestSuite("did_web_resolve")
+
+        @AfterAll
+        fun verifyAllTestsIncluded() {
+            if (testSuite.tests.isNotEmpty()) {
+                println("The following tests were not included or executed:")
+                testSuite.tests.forEach { println(it) }
+                fail("Not all tests were executed! ${testSuite.tests}")
+            }
+        }
+
+        @Test
+        fun test_invalid_did() {
+            testSuite.include()
+
+            val resolutionResult = DidWeb.resolve("something invalid")
+            assertEquals(ResolutionMetadataError.INVALID_DID, resolutionResult.resolutionMetadata.error)
+        }
+
+        @Test
+        fun test_create_then_resolve() {
+            testSuite.include()
+
+            val mockWebServer = MockWebServer()
+            mockWebServer.start()
+
+            val url = mockWebServer.url("")
+
+            val bearerDid = DidWeb.create(url.toString())
+
+            // temporarily removing @context from document, need to rework Document type
+            bearerDid.document.context = null
+
+            mockWebServer.enqueue(
+                MockResponse()
+                    .setResponseCode(200)
+                    .addHeader("Content-Type", "application/json")
+                    .setBody(Json.stringify(bearerDid.document))
+            )
```

Comment:
We could stricten this test by only returning did doc from `/.well-known/did.json `. If you wanna go bananas, we could add a test case for when did uri specifies the path

Reply:
yeah, I agree with both of those items! but for a later time 😅